db/model/subject.js

```diff
@@ -367,7 +367,13 @@ module.exports = function subject(seq, dataTypes) {
               redisOps.addKey(subjectType, inst.absolutePath);
               redisOps.hmSet(subjectType, inst.absolutePath, inst.get());
             } else {
-              subjectUtils.removeFromRedis(inst.absolutePath);
+
+              /*
+               * Delete multiple possible entries in the sample master list of
+               * index
+               */
+              redisOps.deleteKeys(sampleType, subjectType, inst.absolutePath);
+              redisOps.hmSet(subjectType, inst.absolutePath, inst.get());
```

Comment:
Not sure. Is subAspMap only for samples? If so, on unpublish there are no more samples, so subAspMap should be deleted.